### PR TITLE
feat: add listener function and button actions

### PR DESCRIPTION
**Pull request by Yinkusadel**
### This PR Implement a listener function to manage button actions, perform corresponding calculations and Handle the display of results and it's necessary because it serves as the entry point or initialization function for the calculator application.

**CHANGES MADE**

-  Added a startApp function to initialize the calculator application.
- Declared a calcObject to store the calculator's state, including the current numbers (num1 and num2), the display value (displayValue), and the active operator (operator).
- Attached a click event listener to the button container to respond to button clicks.
- Extracted the clicked value from the event target's value attribute and called the calculate function with the clicked value and the current calculator state (calcObject).
- Updated the user input display with the result from the calculate function.

**Here is the link to the projects  [requirement page](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/01-js-calculator-app.md#milestone-6-calculate-function)**

### CODE REVIEW

File: js/calculate.js [hunk 1]
```diff
@@ -66,6 +68,22 @@ const calculate = (input, calcObject) => {
         operator,
       };
 
+    case '.':
+      currentNum = operator !== null ? num2 : num1;
+      hasDecimal = currentNum !== null && currentNum.includes('.');
+
+      if (!hasDecimal) {
```
Comment (teckcipher) [inline js/calculate.js]: Can you please handle positive can first? For example use `operator === null` instead of `operator !== null` and use `if (hasDecimal) {` instead of `if (!hasDecimal) {`

File: js/calculate.js [hunk 2]
```diff
@@ -66,6 +68,22 @@ const calculate = (input, calcObject) => {
...
+        const updatedNum = currentNum === null ? '0.' : `${currentNum}.`;
+        return {
+          num1: operator !== null ? num1 : updatedNum,
+          num2: operator !== null ? updatedNum : num2,
```
Comment (teckcipher) [inline js/calculate.js]: same here

File: js/calculate.test.js [hunk 3]
```diff
@@ -472,3 +472,119 @@ describe('when dividing by zero', () => {
     },
   );
 });
+
+describe('when dividing by zero', () => {
```
Comment (teckcipher) [inline js/calculate.test.js]: Please fix this title

**Review by teckcipher — CHANGES_REQUESTED** [review]
Here are some edge cases you can fix along with the other review comments 

image

File: index.html [hunk 4]
```diff
@@ -52,7 +52,7 @@ <h1 class="calc-text">calc</h1>
             </div>
 
             <div class="user-input-display d-flex justify-center">
-              <div class="user-input d-flex align-center">INVALID OPERATOR</div>
+              <div class="user-input d-flex align-center"></div>
```
Comment (teckcipher) [inline index.html]: Let's use `0` instead.

**Review by adejam — APPROVED** [review]
# LGTM

## Nice job...

PR description can be improved though... using the following
-  acceptance criteria of the task
- adding steps to test
- good format with headers differentiable from paragraphs
- having detailed description of why the task is necessary

**Review by teckcipher — APPROVED** [review]
